python/lsst/pipe/tasks/postprocess.py

```diff
@@ -637,3 +675,102 @@ def writeMetadata(self, dataRef):
         """No metadata to write.
         """
         pass
+
+
+class TransformSourceTableConfig(TransformCatalogBaseConfig):
+    pass
+
+
+class TransformSourceTableTask(TransformCatalogBaseTask):
+    _DefaultName = "transformSourceTable"
+    ConfigClass = TransformSourceTableConfig
+
+    inputDataset = 'source'
+    outputDataset = 'sourceTable'
+
+    def writeMetadata(self, dataRef):
+        """No metadata to write.
+        """
+        pass
+
+    def writeConfig(self, butler, clobber=False, doBackup=True):
+        """No config to write.
+        """
+        pass
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        print("input dataset", cls.inputDataset,)
+        parser = ArgumentParser(name=cls._DefaultName)
+        parser.add_id_argument("--id", datasetType=cls.inputDataset,
+                               level="sensor",
+                               help="data ID, e.g. --id visit=12345 ccd=0")
+        return parser
+
+
+class VisitDataIdContainer(DataIdContainer):
+
+    def makeDataRefList(self, namespace):
+        """Make self.refList from self.idList
+
+        Generate a list of data references grouped by visit
+        """
+        def ccdPatchRefList(visitId):
+            """Get all possible ccds"""
+            ccds = namespace.butler.queryMetadata('src', ['ccd'], dataId={'visit': visitId})
+            return [namespace.butler.dataRef(datasetType=self.datasetType,
+                                             visit=visitId,
+                                             ccd=ccd) for ccd in ccds]
+        # Group by visits
+        visitRefs = defaultdict(list)  # Data references for each tract
+        for dataId in self.idList:
+            if "visit" in dataId:
+                visitId = dataId["visit"]
+                if "ccd" in dataId:
+                    visitRefs[visitId].append(namespace.butler.dataRef(datasetType=self.datasetType,
+                                                                       visit=visitId, ccd=dataId['ccd']))
+                else:
+                    visitRefs[visitId] += ccdPatchRefList(visitId)
+        outputRefList = []
+        for refList in visitRefs.values():
+            existingRefs = [ref for ref in refList if ref.datasetExists()]
+            outputRefList.append(existingRefs)
+
+        self.refList = outputRefList
+
+
+class ConsolidateSourceTableConfig(pexConfig.Config):
+    pass
+
+
+class ConsolidateSourceTableTask(CmdLineTask):
+    """Write patch-merged source tables to a tract-level parquet file
+    """
+    _DefaultName = "consolidateSourceTable"
+    ConfigClass = ConsolidateObjectTableConfig
+
+    inputDataset = 'sourceTable'
+    outputDataset = 'sourceTable_visit'
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = ArgumentParser(name=cls._DefaultName)
+
+        parser.add_id_argument("--id", cls.inputDataset,
+                               help="data ID, e.g. --id visit=12345",
```

Comment:
Do these help statements need to be updated with more specific examples?

Reply:
What do you have in mind?  `Consolidate` has just `visit=12345` to hint it's per-visit and `Transform` has a `visit=12345 ccd=0` to hint it's per detector.